Framework/PDFExporter/PDFExporter/Categories/UIKit/UITableView+Extension.m

```diff
@@ -0,0 +1,69 @@
+//
+//  UITableView+Extension.m
+//  PDFExporter
+//
+//  Copyright © 2016 3Pillar Global. All rights reserved.
+//
+
+#import "UITableView+Extension.h"
+#import "UIView+StatePersistance.h"
+
+@interface UITableViewPersistenceState : UIViewPersistenceState
+
+@property (nonatomic) CGPoint contentOffset;
+
+@end
+
+@implementation UITableViewPersistenceState
+
+@end
+
+@implementation UITableView (Extension)
+
+- (NSArray *)drawingSubviewsForRect:(CGRect)rect {
+    NSMutableArray *drawingSubviews = [NSMutableArray new];
+    if (self.backgroundView) {
+        [drawingSubviews addObject:self.backgroundView];
+    }
+    if (self.contentSize.height > CGRectGetHeight(self.frame)) { // modify contentOffset only if there are invisible cells
+        self.contentOffset = CGPointMake(0, CGRectGetMinY(rect));
+    }
+    [self layoutIfNeeded];
+    [drawingSubviews addObjectsFromArray:[self headerAndFooterViews]];
+    [drawingSubviews addObjectsFromArray:self.visibleCells];
+    return drawingSubviews;
+}
+
+- (NSArray *)headerAndFooterViews {
+    NSMutableArray *headerAndFooterViews = [NSMutableArray new];
+    for (NSUInteger section = 0; section < self.numberOfSections; ++section) {
```

Comment:
section++ or ++section does the same thing. The difference is that ++section doesn't cache old value. It's a C optimization. These days compiler analyzer is smart enough and will do this kind of optimizations. I know that I shouldn't worry about this by I've been writing this way for loops since I started writing Objective-C code.